A6.md

```diff
@@ -0,0 +1,465 @@
+
+In some cases, gRPC can guarantee that a request has never been seen by the server application logic. These cases will be transparently retried by gRPC, as detailed [here](#transparent-retries).
+
+Lastly, information about number of retry attempts will be exposed to the client and server applications through metadata. Find more details [here](#exposed-retry-metadata).
+
+### Detailed Design
+
+#### Retry Policy Capabilities
+
+Retry policies support configuring the maximum number of retries, the parameters for exponential backoff, and the set of retryable status codes, as follows:
+
+```
+"retryPolicy": {  
+  "maxRetryAttempts": 3,
+  "initialBackoffMs": 100,
+  "maxBackoffMs": 1000,
+  "backoffMultiplier": 2,
+  "retryableStatusCodes": [  
+    "UNAVAILABLE"
+  ]
+}
+```
+Each of these configuration options is detailed in its own section below.
+
+##### Maximum Number of Retries
+
+`maxRetryAttempts` specifies the maximum number of retry attempts per original RPC.
+
+gRPC's call deadlines apply to the entire chain of original RPC and retry attempts. The time limit begins when the original RPC is issued to the gRPC client library. For example, if the specified deadline is 500ms and after 500ms only one retry has been attempted, the call is cancelled even if `maxRetryAttempts` is set to two or greater.
+
+##### Exponential Backoff
+
+The `initialBackoffMs`, `maxBackoffMs`, and `backoffMultiplier` parameters determine the randomized delay before retry attempts.
+
+The initial retry attempt will occur at `random(0, initialBackoffMs)`. In general, the `n`-th attempt will occur at `random(0, min(initialBackoffMs*backoffMultiplier**(n-1), maxBackoffMs))`.
+
+##### Retryable Status Codes
+
+When gRPC receives a non-OK response status from a server, this status is checked against the set of retryable status codes in `retryableStatusCodes` to determine if a retry attempt should be made.
+
+In general, only status codes that indicate the service did not process the request should be retried. However, a more aggressive set of parameters can be specified when the service owner knows that a method is idempotent, or safe to be processed more than once by the server. For example, if an RPC to a delete user method fails with an `INTERNAL` error code, it's possible that the user was already deleted before the failure occurred. If the method is idempotent then retrying the call should have no adverse effects. However, it is entirely possible to implement a non-idempotent delete method that has adverse side effects if called multiple times, and such a method should not be retried unless the error codes guarantee the original RPC was not processed. It is up to the service owner to pick the correct set of retryable status codes given the semantics of their service. gRPC retries do not provide a mechanism to specifically mark a method as idempotent.
+
+#### Hedging Policy
```

Comment:
Can we rename hedging to something else? When I was first reading this I thought it meant one: 
send X RPCs immediately and return the call once the first one completes, cancelling the others.